### PR TITLE
Add action based on travis file

### PR DESCRIPTION
**Pull request by vthemelis**
@sipi, if you're happy with this, could you configure the project according to https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository after merge?

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: Setup OS dependencies
+      run: |-
+        sudo apt-get update
+        sudo apt-get install -yy \
+          libnss3 \
+          libatk-bridge2.0-0 \
+          libgtk-3-dev \
+          libasound2 \
+          xvfb
```
Comment (zetashift) [inline .github/workflows/main.yaml:25]: Whats all this for?

**Review by zetashift — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (vthemelis) [reply .github/workflows/main.yaml:25]: For running unit tests in headless mode. 

Basically installed stuff until the testing command stopped complaining about missing shared libraries

**Review by zetashift — APPROVED** [review]
This LGTM, but CI/CD isn't exactly my expertise @sipi

**Comment by vthemelis** [comment]
This now builds on my fork fine. The tests are failing due to a previously introduced regression though.

**Comment by nanoni17728** [comment]
@vthemelis any reason you decided to close your merge request?

**Comment by vthemelis** [comment]
> @vthemelis any reason you decided to close your merge request?

It just didn't seem like there's interest.

**Pull request closed without merging**